src/PostgREST/Error.hs

```diff
@@ -161,46 +195,52 @@ instance JSON.ToJSON PgError where
 
 instance JSON.ToJSON SQL.UsageError where
   toJSON (SQL.ConnectionError e) = JSON.object [
-    "code"    .= ("" :: Text),
+    "code"    .= ConnectionErrorCode00,
     "message" .= ("Database connection error. Retrying the connection." :: Text),
-    "details" .= (T.decodeUtf8With T.lenientDecode $ fromMaybe "" e :: Text)]
+    "details" .= (T.decodeUtf8With T.lenientDecode $ fromMaybe "" e :: Text),
+    "hint"    .= JSON.Null]
   toJSON (SQL.SessionError e) = JSON.toJSON e -- SQL.Error
 
 instance JSON.ToJSON SQL.QueryError where
   toJSON (SQL.QueryError _ _ e) = JSON.toJSON e
 
 instance JSON.ToJSON SQL.CommandError where
-  toJSON (SQL.ResultError (SQL.ServerError c m d h)) = case BS.unpack c of
-    'P':'T':_ -> JSON.object [
-        "details" .= fmap T.decodeUtf8 d,
-        "hint"    .= fmap T.decodeUtf8 h]
-
-    _         -> JSON.object [
+  toJSON (SQL.ResultError (SQL.ServerError c m d h)) = JSON.object [
```

Comment:
So we are fixing https://github.com/PostgREST/postgrest/issues/2176 as well right? This should be added to the CHANGELOG.